app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -904,26 +904,24 @@ class HomeActivityTest {
     )
     logIntoUserTwice()
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId1)).use {
-      it.onActivity { activity ->
-        testCoroutineDispatchers.runCurrent()
-        scrollToPosition(position = 1)
-        onView(
-          atPositionOnView(
-            recyclerViewId = R.id.home_recycler_view,
-            position = 0,
-            targetViewId = R.id.promoted_story_list_recycler_view
-          )
-        ).check { _, _ ->
-          val promotedStoryCard =
-            activity.findViewById<LessonThumbnailImageView>(R.id.lesson_thumbnail)
-          val pixels = promotedStoryCard?.width?.toFloat()
-          val expectedSize = TypedValue.applyDimension(
-            TypedValue.COMPLEX_UNIT_SP,
-            280F,
-            context.resources.displayMetrics
-          )
-          assertThat(pixels).isEqualTo(expectedSize)
-        }
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.home_recycler_view,
+          position = 1,
+          targetViewId = R.id.promoted_story_list_recycler_view
+        )
+      ).check { view, _ ->
+        val promotedStoryCard =
+          view.findViewById<LessonThumbnailImageView>(R.id.lesson_thumbnail)
+        val promotedStoryCardWidth = promotedStoryCard?.width?.toFloat()
+        val expectedWidthInSP = TypedValue.applyDimension(
```

Comment:
Isn't this actually in pixels?

Reply:
Sorry to clarify, the returned unit from applyDimension _has_ to be pixels, here, otherwise the comparison wouldn't work (since a view's width is always provided in pixels). I think we need to rename this value since it's unit isn't sp.

Reply:
Renamed it `expectedWidthInPixels`